File: springboot_export_csv_apache_commons/src/main/java/com/example/demo/controller/EmployeeController.java
package com.example.demo.controller;

import javax.servlet.http.HttpServletResponse;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

import com.example.demo.service.EmployeeService;

@Controller
public class EmployeeController {

	@Autowired
	EmployeeService employeeService;

	@GetMapping("/export-employees")
	public ResponseEntity<InputStreamResource> exportCSV(HttpServletResponse response) throws Exception {

		String filename = "employees.csv";
		InputStreamResource file = new InputStreamResource(employeeService.load());

		return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + filename)
				.contentType(MediaType.parseMediaType("application/csv")).body(file);

	}
}

File: kotlin-spring-main/kotlin_springboot_apachecommans_exportcsv/src/main/kotlin/com/knf/dev/controller/EmployeeController.kt
package com.knf.dev.controller

import org.springframework.http.HttpHeaders

import org.springframework.http.ResponseEntity

import org.springframework.core.io.InputStreamResource

import javax.servlet.http.HttpServletResponse

import org.springframework.web.bind.annotation.GetMapping

import com.knf.dev.service.EmployeeService

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.stereotype.Controller


@Controller
class EmployeeController {
    @Autowired
    var employeeService: EmployeeService? = null
    @GetMapping("/export-employees")
    @Throws(Exception::class)
    fun exportCSV(response: HttpServletResponse?): ResponseEntity<InputStreamResource> {
        val filename = "employees.csv"
        val file = InputStreamResource(employeeService!!.load())
        return ResponseEntity.ok().header(
            HttpHeaders.CONTENT_DISPOSITION,
            "attachment; filename=$filename"
        )
            .contentType(MediaType.parseMediaType("application/csv")).body(file)
    }
}
